plugins/pulp_ostree/plugins/importers/steps.py

```diff
@@ -108,16 +108,29 @@ def process_main(self, item=None):
         Add/update the remote summary information in the
         repository scratchpad.
         """
-        repository = self.get_repo()
+        repository = self.get_repo().repo_obj
         lib_repository = lib.Repository(self.parent.storage_dir)
         remote = lib.Remote(self.parent.repo_id, lib_repository)
+        refs = [r.dict() for r in remote.list_refs()]
+        map(self.convert_metadata_dict, refs)
         repository.scratchpad.update({
             constants.REMOTE: {
-                constants.SUMMARY: [r.dict() for r in remote.list_refs()]
+                constants.SUMMARY: refs
             }
         })
         repository.save()
 
+    @staticmethod
+    def convert_metadata_dict(ref):
+        """
+        Converts the metadata dict part of a ref dict so that keys do not include
+        dots.
+
+        :param ref: a dictionary retrieved with lib.Remote().list_refs()
+        :type  ref: dict
+        """
+        ref['metadata'] = dict((k.replace('.', '-'), v) for k, v in ref['metadata'].items())
```

Comment:
This addresses the same validation problem as the new ``validate`` method on the field subclass above.